eng/common/TestResources/New-TestResources.ps1

```diff
@@ -723,29 +723,27 @@ try {
         Log $msg
 
         $deployment = Retry {
-            $lastDebugPreference = $DebugPreference
-            try {
-                if ($CI) {
-                    $DebugPreference = 'Continue'
-                }
-                New-AzResourceGroupDeployment -Name $BaseName -ResourceGroupName $resourceGroup.ResourceGroupName -TemplateFile $templateFile.jsonFilePath -TemplateParameterObject $templateFileParameters -Force:$Force
-            } catch {
-                Write-Output @'
+            New-AzResourceGroupDeployment `
```

Comment:
Based on the old code it would seem as though this might throw exceptions if it failed. Do you know if that is true or not?

Reply:
Yes, it could throw or return an error- `try...catch...finally` in powershell works for both. The idea was to reset the `$DebugPreference`, which we don't need anymore. Why it kept going I don't know - might've been an oversight after all the changes in this area over the years.

Reply:
It does throw still, which gets caught by the `Retry` function handler instead. I was confused at first as well (though it was my old code so...)

Reply:
I guess Retry only exits with a Write-Error, so perhaps we should actually re-throw in case the ErrorActionPreference is different for local users?

Reply:
That could work.

There are some ill-behaved cmdlets like this that actual throw instead of write error, including the first-party `Invoke-WebRequest` (which just bit me today; I had no idea it wasn't...proper).